### DISCUSSION TITLE
Omit quotes string

### DISCUSSION THREAD

**Discussion by Shatur** [pest-parser/pest#947] [Q&A] [answered]
I trying to omit string quotes ([playground](https://pest.rs/?g=N4Ig5gTghgtjURALhAMwK4DsDGAXAlgPaYAEAvCQALAkA6ItjDIJAfiQG5QA26ApmzoMm9EgF9amLrwEVqJAIIBlAMIBJNQH0FAGQAKACQUBqcZPOYA6gbUAVAKJK9ClffIlNNUaLEgANCD4mAAO6LjIzAr0IGJAA#editor)). If I make `function` rule silent it will work, but I need it to be atomic too. Is it possible to create a rule that is atomic and silent?

**Comment by NyalephTheCat** [comment]
You can technically make two rules, one inner being atomic, the outer being silent... I'm not sure if it is possible to do better.

But, for what you want, [this ](https://pest.rs/?g=N4Ig5gTghgtjURALhAMwK4DsDGAXAlgPaYAEAvCQALAkCO6huApgM4kB%2BJAblADbpMOdBszYBfADqYe-QRWokAggGUAwgEl1AfUUAZAAoAJRQGoSkzFPqNW5ElpoSQEl85DmpAdUPqAKgFFlfUVVfzsHEidI9zEQABoQfEwAB3RcZDdFJxAxIA#editor) should be good enough

**Reply by Shatur** [reply]
> You can technically make two rules, one inner being atomic, the outer being silent...

Oh, got it, thanks!

> But, for what you want, [this ](https://pest.rs/?g=N4Ig5gTghgtjURALhAMwK4DsDGAXAlgPaYAEAvCQALAkCO6huApgM4kB%2BJAblADbpMOdBszYBfADqYe-QRWokAggGUAwgEl1AfUUAZAAoAJRQGoSkzFPqNW5ElpoSQEl85DmpAdUPqAKgFFlfUVVfzsHEidI9zEQABoQfEwAB3RcZDdFJxAxIA#editor)should be good enough

But the result contains quotes, I trying to exclude them. I tried this approach myself before.

**Reply by Shatur** [reply]
> You can technically make two rules, one inner being atomic, the outer being silent...

This approach will omit spaces that are inside the string, [code](https://pest.rs/?g=N4Ig5gTghgtjURALhARwK4HsAuBTAJgPoDO2EAlgHZgAEAvDYcDQDogsfsg0B%2BN0AdxJkqtPmw4TuAXxaVBwitXo0AAswCCAZQDCAST2ENAGQAKACQ0BqGrMpyA6ub0AVAKJbTGnW5VNW3Gy2IAA0IFQADujYyFw0GgEg0kA#editor).

**Reply by NyalephTheCat** [reply, marked as answer]
Oh, right. Then you'll have to go through the generated tokens and just fetch the value one.... I'm pretty sure this might be the only solution right now. It might be worth making an issue suggesting it. Unless I forgot the way to do it

**Reply by Shatur** [reply]
Got it, thanks!
